packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4001,6 +4001,31 @@ class IndexedStack extends Stack {
       ..alignment = alignment
       ..textDirection = textDirection ?? Directionality.maybeOf(context);
   }
+
+  @override
+  MultiChildRenderObjectElement createElement() {
+    return _IndexedStackElement(this);
+  }
+}
+
+class _IndexedStackElement extends MultiChildRenderObjectElement {
+  _IndexedStackElement(IndexedStack super.widget);
+
+  @override
+  IndexedStack get widget => super.widget as IndexedStack;
+
+  @override
+  void debugVisitOnstageChildren(ElementVisitor visitor) {
+    final int? index = widget.index;
+    if (index == null) {
+      return super.debugVisitOnstageChildren(visitor);
+    } else {
+      final Iterator<Element> onlyOnstageChild = children.skip(index).iterator;
+      if (onlyOnstageChild.moveNext()) {
+        visitor(onlyOnstageChild.current);
+      }
```

Comment:
I had to special-case the scenario where `index` is out of bounds because there are no children (the default when constructing an `IndexedStack()`), but `elementAt` is much more readable. Thanks for the suggestion!